animated_sprite.cpp

```diff
@@ -0,0 +1,113 @@
+#include "animated_sprite.h"
+#include <iostream>
+
+// AnimatedSprite()
+AnimatedSprite::AnimatedSprite(std::string sprite_sheet_name,
+                               SDL_Rect first_sprite_coord_location,
+                               SDL_Renderer* renderer,
+                               float time_to_update_sprite,
+                               SDL_Rect destination)
+    : Sprite(sprite_sheet_name, first_sprite_coord_location, renderer),
+      time_to_update(time_to_update_sprite),
+      current_animation_once(false),
+      current_animation(""),
+      frame_index_(0),
+      visible_(true),
+      destination_(destination) {
+  this->SetupAnimations();
+}
+
```

Comment:
this-> is not necessary inside methods in the class. Just "SetupAnimations" is fine, the this-> is implied.